src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java

```diff
@@ -725,8 +728,24 @@ public static boolean handleCollideBlockEvent(final Block block, final net.minec
             }
 
             // TODO: Add target side support
-            final CollideBlockEvent event = SpongeEventFactory.createCollideBlockEvent(frame.currentCause(), (BlockState) state,
-                    ServerLocation.of((org.spongepowered.api.world.server.ServerWorld) world, VecHelper.toVector3d(pos)), direction);
+            final ServerLocation loc = ServerLocation.of((ServerWorld) world, VecHelper.toVector3d(pos));
+            final CollideBlockEvent event;
+            switch (type) {
+                case "move":
```

Comment:
Magic strings like this are a bad idea because you have to change this in multiple places, and you might not know where they all are - very easy to miss.

These strings should be enums instead for nothing more than ease of maintenance's sake.

Reply:
Agreed, this needs to be a enum 100%